### PR TITLE
Added check of integration while finding Fourier Transform

### LINKED ISSUE #12591: Fourier transform of exponential returns 0
When I try to compute the Fourier Transform of exp(x) it returns 0.

     import sympy as sym
     x, y = sym.symbols("x y", real=True)
     fun = sym.fourier_transform(sym.exp(x), x, y)
     print(fun)

returns

    0

### PR DESCRIPTION
**Pull request by ashishkg0022**
before this `fourier_transform(sym.exp(x), x, y)` returned `zero`  which was wrong.
so added a check after finding intgegration.
Fixes #12591

### CODE REVIEW

**Comment by ashishkg0022** [comment]
ping @jksuom

**Comment by jksuom** [comment]
It is probable that the integral test should come before the `not F.has(Integral)` line. In some cases F may be wrong when the function is not integrable (`f = 1`, for example).

File: sympy/integrals/transforms.py [hunk 1]
```diff
@@ -1293,6 +1293,11 @@ def _fourier_transform(f, x, k, a, b, name, simplify=True):
     if not F.has(Integral):
         return _simplify(F, simplify), True
 
+    integral_f = integrate(f, (x,-oo,oo))
+    if integral_f is oo or integral_f is -oo or \
```
Comment (jksuom) [inline sympy/integrals/transforms.py]: These can be combined: `integral_f in (oo, -oo, S.NaN)`. Maybe also `integral_f.has(Integral)` should be added.

**Comment by ashishkg0022** [comment]
@jksuom , I was having a doubt. for now should we restrict ourselves to checking limit at `-oo and oo` or finding integral is fine. Are there some functions whose integration from `-oo to oo` does not exist but their Fourier transform is possible to find?

File: sympy/integrals/transforms.py (hunk 1 above)
Comment (ashishkg0022) [reply sympy/integrals/transforms.py]: on adding `integral_f.has(Integral)` a previous test is failing .
```python
# NOTE: the ift comes out in pieces
    assert IFT(1/(a + 2*pi*I*x), x, posk,
            noconds=False) == (exp(-a*posk), True)
```
`AssertionError`

**Comment by jksuom** [comment]
> Are there some functions whose integration from -oo to oo does not exist but their Fourier transform is possible to find?

There are non-integrable functions whose Fourier transform can be defined as a tempered distribution but not as a function. For example, the transform of `f = 1` can be defined as Dirac delta. However, such transformations have not been implemented in SymPy, and the answer it gives, if any, is wrong. So, I think, raising an error in those cases would be reasonable. This should be changed when those transforms are implemented.

File: sympy/integrals/transforms.py [hunk 2]
```diff
@@ -1293,6 +1293,10 @@ def _fourier_transform(f, x, k, a, b, name, simplify=True):
     if not F.has(Integral):
         return _simplify(F, simplify), True
 
+    integral_f = integrate(f, (x,-oo,oo))
```
Comment (jksuom) [inline sympy/integrals/transforms.py]: This would look better with a space after the two last commas.

**Comment by jksuom** [comment]
This looks good to me, thanks!

**Pull request merged**